responsive-test-utils/src/main/java/dev/responsive/kafka/internal/clients/TTDCassandraClient.java

```diff
@@ -42,16 +42,16 @@ public class TTDCassandraClient extends CassandraClient {
   private final ResponsiveStoreRegistry storeRegistry = new ResponsiveStoreRegistry();
   private final TTDMockAdmin admin;
 
-  private final TableFactory<RemoteKVTable> kvFactory;
-  private final TableFactory<RemoteWindowedTable> windowedFactory;
+  private final TableCache<RemoteKVTable> kvFactory;
+  private final TableCache<RemoteWindowedTable> windowedFactory;
 
   public TTDCassandraClient(final TTDMockAdmin admin, final Time time) {
```

Comment:
Another thing that would be nice about generalizing the storage backend at the Client/ClientFactory level (ie having the CassandraClient and MongoClient implement a common client & factory interface vs branching immediately based on enums) is that we could probably throw out a lot of the gross hacks we currently put into the TTD version of the C* client, and just have a separate TTDClient and TTDClientFactory that is completely state-store agnostic.

We could even make it fully pluggable so that users could choose whether to run with the mock client or use one of the real storage backends, eg with a container setup. I think that would be pretty slick

Obviously can/should be a followup project, just thinking out loud here

Reply:
that would be pretty cool indeed! I'll turn this into a ticket so we don't forget about it. #165